awscli/customizations/generatecliskeleton.py

```diff
@@ -133,3 +147,117 @@ def _make_client_call(self, client, operation_name, parameters,
         with Stubber(client) as stubber:
             stubber.add_response(method_name, fake_response)
             return getattr(client, method_name)(**parameters)
+
+
+class _Bytes(object):
+    @classmethod
+    def represent(cls, dumper, data):
+        return dumper.represent_scalar(u'tag:yaml.org,2002:binary', '')
+
+
+class YAMLArgumentGenerator(ArgumentGenerator):
+    _SENTENCE_DELIMETERS_REGEX = re.compile(r'[.:]+')
+    _LINE_BREAK_CHARS = [
+        '\n',
+        '\u2028'
+    ]
+
+    def __init__(self, use_member_names=False, yaml=None):
+        super(YAMLArgumentGenerator, self).__init__(
+            use_member_names=use_member_names)
+        self._yaml = yaml
+        if self._yaml is None:
+            self._yaml = YAML()
+
+    def _generate_skeleton(self, shape, stack, name=''):
+        # YAML supports binary, so add in boilerplate for that instead of
+        # putting in None. Here were' using a custom type so that we can ensure
+        # we serialize it correctly on python 2 and to make the
+        # serialization output more usable on python 3.
+        if shape.type_name == 'blob':
+            return _Bytes()
+        return super(YAMLArgumentGenerator, self)._generate_skeleton(
+            shape, stack, name
+        )
+
+    def _generate_type_structure(self, shape, stack):
+        if stack.count(shape.name) > 1:
+            return self._yaml.map()
+        skeleton = self._yaml.map()
+        for member_name, member_shape in shape.members.items():
+            skeleton[member_name] = self._generate_skeleton(
+                member_shape, stack, name=member_name)
+            is_required = member_name in shape.required_members
+            self._add_member_comments(
+                skeleton, member_name, member_shape, is_required)
+        return skeleton
+
+    def _add_member_comments(self, skeleton, member_name, member_shape,
+                             is_required):
+        comment_components = []
+        if is_required:
+            comment_components.append('[REQUIRED]')
+        comment_components.append(
+            self._get_comment_content_from_documentation(member_shape)
+        )
+        if getattr(member_shape, 'enum', None):
```

Comment:
What do you think about putting the enum values before the documentation? It's probably the most useful bit and having it before the description sentence would make it easier to see. Otherwise you have to scroll past the description sentence to see the values you can enter.